packages/core/lib/span.ts

```diff
@@ -64,6 +64,17 @@ export function spanToJson (span: SpanEnded, clock: Clock): DeliverySpan {
   }
 }
 
+export function spanEndedToSpan (span: SpanEnded): Span {
+  return {
+    id: span.id,
+    traceId: span.traceId,
+    samplingRate: span.samplingRate,
```

Comment:
Possibly beyond the scope of this PR, but it feels like these properties should be made immutable in the same way as in `SpanFactory.toPublicApi()`

Also, should we expose the span name here and in the public Span API (also as an immutable property)? If you're writing an onSpanEnd callback it seems like that would be useful to know.

Reply:
Yeah. `name` should definitely be public (and immutable)